Line_2 $$ private JasperPrint doFillReport(JasperReport report, Map<String, Object> model, DataSource ds) throws Exception -->Line_6 $$ Connection con = ds.getConnection()[ FD ]
Line_2 $$ private JasperPrint doFillReport(JasperReport report, Map<String, Object> model, DataSource ds) throws Exception -->Line_3 $$ if (logger.isDebugEnabled()) [ CD ]
Line_6 $$ Connection con = ds.getConnection()-->Line_11 $$ con.close()[ FD ]
Line_6 $$ Connection con = ds.getConnection()-->Line_8 $$ return JasperFillManager.fillReport(report, model, con)[ FD ]
Line_2 $$ private JasperPrint doFillReport(JasperReport report, Map<String, Object> model, DataSource ds) throws Exception -->Line_8 $$ return JasperFillManager.fillReport(report, model, con)[ FD ]
Line_2 $$ private JasperPrint doFillReport(JasperReport report, Map<String, Object> model, DataSource ds) throws Exception -->Line_6 $$ Connection con = ds.getConnection()[ CD ]
Line_3 $$ if (logger.isDebugEnabled()) -->Line_4 $$ logger.debug("Filling report using JDBC DataSource [" + ds + "]")[ CD ]
